### DISCUSSION TITLE
Moving from ASM6 to ca65 - how to handle with only one pass?

### DISCUSSION THREAD

**Discussion by MartsINY1** [cc65/cc65#2425] [Q&A] [answered]
Here is the content of my file compiled by ca.65
```
.segment "CODE"

.include "..\..\_Mega_Man_Engine\__HEADER\main__header.s"
.include "..\_CHR_BANKS\main__CHR_ROM__SPRITES.s"
```
Here is content of 
> main__header.s

```;5: Size of CHR ROM in 8 KB units
.segment "HEADER"
.byte <CHR__ROM__BANK__CURRENT
```

Here is content of 

> main__CHR_ROM__SPRITES.s

```CHR__ROM__BANK__CURRENT .set $0000

.segment "MARIO"
.incbin "..\..\_Mega_Man_Engine\_CHR_BANKS\DATA\PLAYABLE_CHARACTERS\mario.chr"
marioBank .set CHR__ROM__BANK__CURRENT
CHR__ROM__BANK__CURRENT .set (CHR__ROM__BANK__CURRENT + $01)

.segment "LUIGI"
.incbin "..\..\_Mega_Man_Engine\_CHR_BANKS\DATA\PLAYABLE_CHARACTERS\luigi.chr"
luigiBank .set CHR__ROM__BANK__CURRENT
CHR__ROM__BANK__CURRENT .set (CHR__ROM__BANK__CURRENT + $01)
```

Here is the config file
```
MEMORY {
    HDR:					start = $0000,  size = $0010, type = ro, file = %O, fill = yes, fillval = $EA;
    PRG:					start = $8000,  size = $100000, type = ro, file = %O, fill = yes, fillval = $EA;
    CHR:					start = $00000,  size = $100000, type = ro, file = %O, fill = yes, fillval = $EA;
}

SEGMENTS {
    HEADER:   load = HDR, type = ro;
    CODE:     load = PRG, type = ro,  start = $8000;
    marioBank:	load = CHR, 			type = ro;
    luigiBank :	load = CHR, 			type = ro;
}
```

My problem is that I want the part of header that is:
`.byte <CHR__ROM__BANK__CURRENT`

To output value once it has been incremented and initiated in
> main__CHR_ROM__SPRITES.s

Now, a solution is to just switch code of my file compiled by ca65 to be:
```
.segment "CODE"

.include "..\_CHR_BANKS\main__CHR_ROM__SPRITES.s"
.include "..\..\_Mega_Man_Engine\__HEADER\main__header.s"
```

But that solution is not valid since I have many other elements code elements to convert from ASM6 to ca65 and it won't be possible with those - my goal is to be able to swap every include without needing to care about the order. ASM6 was perfect for that since it made many passes - all I had to do was to have it pass until I had same result twice in a row. Any solution that can allow me to do that would be fine for me.

**Comment by mrdudz** [comment, marked as answer]
It seems this is trivial to solve:

- assemble main__header.s and main__CHR_ROM__SPRITES.s separately to object files
- in  main__header. .import the the CHR__ROM__BANK__CURRENT symbol
- in  main__CHR_ROM__SPRITES.s .export the the CHR__ROM__BANK__CURRENT symbol
- link the two files together

**Comment by MartsINY1** [comment]
Thanks a lot this worked! I actually found out a simpler way though, I just use the variable and it gets defined later.
However what you told me showed me many new possibilities, I didn't know about all those stuff I could do!! Thanks a lot!!
